server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSCsdSpOptionsCheckUtility.java

```diff
@@ -0,0 +1,341 @@
+            checkHasMandatoryOptions(ctx.RESID(), ctx, "RESID");
+            checkCvda(ctx.cics_csd_cvda().get(0));
+        } else if (!ctx.RESID().isEmpty()) {
+            checkHasMandatoryOptions(ctx.cics_csd_cvda(), ctx, CVDA_OPTS);
+        }
+    }
+    private void checkGetNextGroup(CICSParser.Cics_csd_getnextgroupContext ctx) {
+        checkHasMandatoryOptions(ctx.GETNEXTGROUP(), ctx, "GETNEXTGROUP");
+        checkHasMandatoryOptions(ctx.GROUP(), ctx, "GROUP");
+    }
+    private void checkGetNextList(CICSParser.Cics_csd_getnextlistContext ctx) {
+        checkHasMandatoryOptions(ctx.GETNEXTLIST(), ctx, "GETNEXTLIST");
+        checkHasMandatoryOptions(ctx.LIST(), ctx, "LIST");
+    }
+    private void checkGetNextRsrce(CICSParser.Cics_csd_getnextrsrceContext ctx) {
+        checkHasMandatoryOptions(ctx.GETNEXTRSRCE(), ctx, "GETNEXTRSRCE");
+        checkHasMandatoryOptions(ctx.RESTYPE(), ctx, "RESTYPE");
+        checkHasMandatoryOptions(ctx.RESID(), ctx, "RESID");
+        checkHasMandatoryOptions(ctx.GROUP(), ctx, "GROUP");
+        checkHasMutuallyExclusiveOptions("ATTRIBUTES or SET", ctx.ATTRIBUTES(), ctx.ATTRIBUTES(), ctx.SET());
+        if (!ctx.ATTRLEN().isEmpty() && (ctx.ATTRIBUTES().isEmpty() && ctx.SET().isEmpty())) {
+            throwException(
+                    ErrorSeverity.ERROR, getLocality(ctx), "Missing required option for ATTRLEN: ATTRIBUTES or SET", "");
+        }
+    }
+    private void checkInquireGroup(CICSParser.Cics_csd_inquiregroupContext ctx) {
+        checkHasMandatoryOptions(ctx.INQUIREGROUP(), ctx, "INQUIREGROUP");
+        checkHasMandatoryOptions(ctx.GROUP(), ctx, "GROUP");
+    }
+    private void checkInquireList(CICSParser.Cics_csd_inquirelistContext ctx) {
+        checkHasMandatoryOptions(ctx.INQUIRELIST(), ctx, "INQUIRELIST");
+        checkHasMandatoryOptions(ctx.LIST(), ctx, "LIST");
+    }
+    private void checkInquireRsrce(CICSParser.Cics_csd_inquirersrceContext ctx) {
+        checkHasMandatoryOptions(ctx.INQUIRERSRCE(), ctx, "INQUIRERSRCE");
+        checkHasMandatoryOptions(ctx.RESID(), ctx, "RESID");
+        checkHasMandatoryOptions(ctx.GROUP(), ctx, "GROUP");
+        checkHasMandatoryOptions(ctx.cics_csd_cvda(), ctx, CVDA_OPTS);
+        checkHasExactlyOneOption("ATTRIBUTES or SET", ctx, ctx.ATTRIBUTES(), ctx.SET());
+        if (!ctx.ATTRLEN().isEmpty() && (ctx.ATTRIBUTES().isEmpty() && ctx.SET().isEmpty())) {
+            throwException(
+                    ErrorSeverity.ERROR, getLocality(ctx), "Missing required option for ATTRLEN: ATTRIBUTES or SET", "");
+        }
+    }
+    private void  checkInstall(CICSParser.Cics_csd_installContext ctx) {
+        checkHasMandatoryOptions(ctx.INSTALL(), ctx, "INSTALL");
+        checkHasMandatoryOptions(ctx.GROUP(), ctx, "GROUP");
+        if (!ctx.cics_csd_cvda().isEmpty()) {
+            checkCvda(ctx.cics_csd_cvda().get(0));
+            checkHasMandatoryOptions(ctx.RESID(), ctx, "RESID");
```

Comment:
RESTYPE and RESID - all or nothing.

Reply:
RESID can be used with resources 
![image](https://github.com/user-attachments/assets/08b9fad1-666f-4657-94c5-d5e45d5450ec)

Reply:
exactly - atomservice is an RESTYPE alias... all or nothing for cics_csd_cvda and RESID.